### PR TITLE
Changes necessary for the front-end design

### LINKED ISSUE #65: Changes necessary for the front-end design
- delete from product or make it unrequired (product condition - shipping option -post type ' is event ).
- delete from register (age -gender - refugee -nationality)

### PR DESCRIPTION
**Pull request by khaldarov**
### Might still require some changes; please check if your previous work does comply with the new version's rules.

 - Fixes `userSchema`.
- Fixes `productSchema`.
- Fixes `register` and documentation.
- Fixes `addNewProduct` and documentation.
- Fixes a bug that prevented products of a deleted user to be automatically deleted.

### CODE REVIEW

File: src/controllers/auth.js [hunk 1]
```diff
@@ -5,17 +5,7 @@ const constants = require('../lib/constants');
...
-    lastName,
-    email,
-    phoneNumber,
-    age,
-    gender,
-    nationality,
-    refugee,
-    password0,
-  } = req.body;
+  const { firstName, lastName, email, phoneNumber, password0 } = req.body;
```
Comment (awiednoor) [inline src/controllers/auth.js:11]: so now we don't have these fields when registering, but they can be kept during updating the user? or are we removing them from everywhere?

File: src/models/product.js [hunk 2]
```diff
@@ -34,20 +34,22 @@ const productSchema = new mongoose.Schema(
...
       enum: constants.ENUM_SHIPPING_OPTION,
-      required: true,
+      required: false,
     },
     postType: {
       type: String,
       enum: constants.ENUM_POST_TYPE,
-      required: true,
+      required: false,
+      defauult: 'Donate',
```
Comment (khatibAmjad) [inline src/models/product.js]: typo here

**Review by khatibAmjad — CHANGES_REQUESTED** [review]

**Comment by halitbatur** [comment]
@khaldarov can we get this fixed so that we can merge it

**Review by khatibAmjad — APPROVED** [review]

**Review by okurezgi — APPROVED** [review]

**Review by awiednoor — APPROVED** [review]
Great work Adnan

File: src/controllers/auth.js (hunk 1 above)
Comment (khaldarov) [reply src/controllers/auth.js:11]: We do not require them when registering, but users can add them later on if they want. It also might be kept as only a BE feature.

**Pull request merged**